cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -193,7 +185,25 @@ public ProgramScheduleStatus scheduleState(ProgramId program, SchedulableProgram
                                              String scheduleName)
     throws SchedulerException, NotFoundException {
     Scheduler scheduler = getSchedulerForSchedule(program, scheduleName);
-    return scheduler.scheduleState(program, programType, scheduleName);
+    ProgramScheduleStatus status;
+    try {
+      status = scheduler.scheduleState(program, programType, scheduleName);
+    } catch (NotFoundException e) {
```

Comment:
is similar kind of handling needed for other methods, too? What if I try to delete a schedule before the scheduler is up? It seems that it is safer to throw ServiceNotAvailableException for all operations, if the scheduler is not started fully.